src/js/containers/search/table/ResultsTableContainer.jsx

```diff
@@ -116,6 +120,10 @@ export class ResultsTableContainer extends React.Component {
                 this.performSearch();
             }
         }
+        else if (prevProps.columnVisibility !== this.props.columnVisibility) {
+            // Visible columns have changed
+            this.showColumns(this.props.meta.tableType);
```

Comment:
Now performs a new search when columns are added or removed, requesting only the fields pertaining to visible columns.